.github/workflows/static_code_analysis.yaml

```diff
@@ -15,24 +15,14 @@ jobs:
       - uses: pre-commit/action@v3.0.0
         env:
           SKIP: no-commit-to-branch
+      - name: ruff
+        uses: chartboost/ruff-action@v1
       - name: black
         run: |
           black --check .
-      - name: isort
```

Comment:
I saw that ruff also checks the import sorting. However, will it also correct it? I.e. is it acting as a linter or a formatter here? If the it's only linting, we might still consider using isort.